scrapy/extensions/feedexport.py

```diff
@@ -483,6 +483,14 @@ def _get_uri_params(self, spider, uri_params, slot=None):
         params['time'] = utc_now.replace(microsecond=0).isoformat().replace(':', '-')
         params['batch_time'] = utc_now.isoformat().replace(':', '-')
         params['batch_id'] = slot.batch_id + 1 if slot is not None else 1
-        uripar_function = load_object(uri_params) if uri_params else lambda x, y: None
-        uripar_function(params, spider)
-        return params
+        uripar_function = load_object(uri_params) if uri_params else lambda x, y: x
+        new_params = uripar_function(params, spider)
```

Comment:
I’m just leaving a note that it is technically possible for this change to break code if the user was modifying `params` but returning a dict with a different content from that of `params`. I don’t think this should stop this change, but it may be worth mentioning in the release notes.

Reply:
What if we checked this? Making a shallow copy of the original `params` dict and making sure it didn't change after calling the `uripar_function`. Something like:

```python
original_params = params.copy()
new_params = uripar_function(params, spider)
if new_params is None or original_params != params:
   ...
```

Reply:
I think it makes sense, though then we must document that as well, before people start using `.pop()` on the original to build the new one and things stop behaving as expected. Once we remove support for editing the dictionary in place, we can start allowing people to do that.